tests/unit/aiplatform/test_lro.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2020, Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import mock
+
+from google.api_core import operation
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import lro
+from google.cloud.aiplatform_v1beta1.services.model_service.client import (
+    ModelServiceClient,
+)
+from google.cloud.aiplatform_v1beta1.types import model as gca_model
+from google.cloud.aiplatform_v1beta1.types.model_service import UploadModelResponse
+from google.longrunning import operations_pb2
+from google.protobuf import struct_pb2 as struct
+
+
+TEST_OPERATION_NAME = "test/operation"
+
+
+class AiPlatformResourceNounImpl(base.AiPlatformResourceNoun):
+    client_class = ModelServiceClient
+    _is_client_prediction_client = False
+
+
+def make_operation_proto(name=TEST_OPERATION_NAME, response=None, **kwargs):
+    operation_proto = operations_pb2.Operation(name=name, **kwargs)
+
+    if response is not None:
+        operation_proto.response.Pack(response)
+
+    return operation_proto
+
+
+def make_operation_future(client_operations_responses=None):
+    if client_operations_responses is None:
+        client_operations_responses = [make_operation_proto()]
+
+    refresh = mock.Mock(spec=["__call__"], side_effect=client_operations_responses)
+    refresh.responses = client_operations_responses
+    cancel = mock.Mock(spec=["__call__"])
+    operation_future = operation.Operation(
+        client_operations_responses[0],
+        refresh,
+        cancel,
+        result_type=gca_model.Model,
+        metadata_type=struct.Struct,
+    )
+
+    return operation_future
+
+
+def test_constructor():
+    operation_future = make_operation_future()
+    test_lro = lro.LRO(operation_future)
+
+    assert test_lro.operation_future.operation.name == TEST_OPERATION_NAME
+    assert test_lro.operation_future._result_type is gca_model.Model
+    assert test_lro.operation_future.done() is False
+    assert len(test_lro.operation_future._done_callbacks) == 0
+
+
+def test_constructor_with_update():
+    operation_future = make_operation_future()
+    resource_noun_obj = AiPlatformResourceNounImpl()
+    result_key = "name"
+    api_get = mock.Mock(spec=["__call__"])
+    test_lro = lro.LRO(operation_future, resource_noun_obj, result_key, api_get)
+
+    assert len(test_lro.operation_future._done_callbacks) == 1
+
+
+def test_update_resource():
+    expected_result = UploadModelResponse(model="tardigrade")
```

Comment:
looks like Any.Pack() errors because this proto.Message does not have descriptor. does proto plus not handle this? I'm a little confused.

Reply:
or i guess a better question is how do I set operations_pb2.Operation.response.

Reply:
[Proto-plus](https://proto-plus-python.readthedocs.io/en/latest/reference/message.html) wraps the raw python protobuf messages so protobuf Python methods won't work on it directly.

I'm not completely following what you're trying to do. Could I have more context? [This api-core](https://github.com/googleapis/python-api-core/blob/85e14e3f3a6b3e3c3ae9626daf60183f4c05fb6a/google/api_core/operation.py#L116) code sets the response based on the response from the API.

Reply:
ah yes. I'm trying to run [operation.py#L133](https://github.com/googleapis/python-api-core/blob/85e14e3f3a6b3e3c3ae9626daf60183f4c05fb6a/google/api_core/operation.py#L133) during init. In [test_lro#L89](https://github.com/googleapis/python-aiplatform/blob/c9d33e98cc66b10f4f966ea84eb8b4494bca16b7/tests/unit/aiplatform/test_lro.py#L89), I'm trying to do this by:
1. setting google.longrunning.operations_pb2.Operation.done to True
2. assigning something to google.longrunning.operations_pb2.Operation.response

The problem I'm facing is part 2 as shown in [test_lro#L41](https://github.com/googleapis/python-aiplatform/blob/c9d33e98cc66b10f4f966ea84eb8b4494bca16b7/tests/unit/aiplatform/test_lro.py#L41).

Reply:
I've tried:
- `operation_proto.response.Pack(response)`
- `operation_proto.response = response`

I'm probably missing something obvious.

Reply:
Ah makes sense 👍. Since `operation_proto` is a protobuf python object, the underlying `pb`  needs to be assigned. You can convert the `proto.Message` into python protobuf through the `.pb()` class method. 

https://proto-plus-python.readthedocs.io/en/latest/reference/message.html#proto.message.Message.pb


```py
from google.longrunning import operations_pb2
import proto

class Row(proto.Message):
    value = proto.Field(proto.STRING, number=1)

row = Row(value='abc')
operation_proto = operations_pb2.Operation(name="test")
operation_proto.response.Pack(Row.pb(row))
```

```py
>>> operation_proto
name: "test"
response {
  type_url: "type.googleapis.com/Row"
  value: "\n\003abc"
}
```